core/domain/prod_validators.py

```diff
@@ -261,7 +264,10 @@ def _get_external_id_relationships(cls, item):
                 [item.id[:item.id.rfind(base_models.VERSION_DELIMITER)]]),
             base_model_validators.ExternalModelFetcherDetails(
                 'committer_ids',
-                user_models.UserSettingsModel, [item.committer_id])]
+                user_models.UserSettingsModel, [item.committer_id],
+                remove_system_user_ids=True,
+                remove_pseudonymous_ids=True
```

Comment:
In most of the places I see this to be `True` then why do we have `False` as a default value? is it to make it clear in all the places? (not sure, any specific reason?)

Reply:
he remove_system_user_ids and the remove_pseudonymous_ids are set to true only if the attributes contain the user ids and the oppia bot and pseudo user ids need to be avoided. So the default value should be false as not all attributes contain user_ids. According to vojtechjelinek suggestion only specific models with user_ids should allow oppia bot and pseudo ids. This list is mentioned in the description.

Reply:
I think it makes sense to have it to false by default since in new models we do not expect these to be true until we know that they will contain admin or pseudo IDs.